src/lndHealth.ts

```diff
@@ -0,0 +1,40 @@
+import { authenticatedLndGrpc, getWalletInfo } from 'lightning';
+import _ from "lodash";
+import { baseLogger } from "./logger";
+
+const refresh_time = 5000 // ms
+
+// TODO replace with
+// const asyncForever = require('async/forever');
+export const isUpLoop = async ({socket, params}): Promise<void> => {
+  let active
+  try {
+    // @ts-ignore
+    const { lnd } = _.find(params, {socket})
+
+    // will throw if there is an error
+    await getWalletInfo({lnd})
+    active = true
+  } catch (err) {
+    baseLogger.warn({err}, `can't get wallet info from ${socket}`)
+
+    // if we get disconnected, we need to recreate the lnd object
+    const paramIndex = _.findIndex(params, {socket})
+    params[paramIndex] = {
+      ...params[paramIndex],
+      lnd: authenticatedLndGrpc(params[paramIndex]).lnd
+    }
+
+    active = false
+  }
+  _.find(params, {socket}).active = active
+  baseLogger.info({socket, active}, "lnd pulse")
+
+  setTimeout(async function () {
+    // TODO check if this could lead to a stack overflow
+    isUpLoop({socket, params})
```

Comment:
I still have the same question regarding this. From above -
"Can we not just call isUpLoop inside the setTimeout to remove the need for recursion? Maybe I'm missing why there's a necessity for recursion here"

Why can't we just do this -> `setTimeout(() => isUpLoop(), refresh_time)` and have the setTimeout be outside the isUpLoop function?

Reply:
oh right ok.
so what we want to do here is to update the array and set `{active: true}` or `{active: false}` over time depending on whether lnd responds. so we want a regular call.

it's like the liveness probe for kubernetes if you want. 

if lnd1 if offline, then graphql will switch automatically to lnd2. 

does that answer your question?

Reply:
I mixed up setInterval and setTimeout, my bad.

Why not use `setInterval` instead of `setTimeout`? That will remove the need for recursion.